### PR TITLE
improved issue templates

### PR DESCRIPTION
**Pull request by Uzlopak**
This PR has improved issue templates. Inspired by https://github.com/fastify/.github

It would be good to always set a label for new bugs like "needs triage". So we can read the bug report and process further like usual with "needs clarification", "needs repro script", "confirmed bug" etc...

### CODE REVIEW

**Comment by Uzlopak** [comment]
I set the branch as my default branch in my fork. So you can test it.

https://github.com/Uzlopak/mongoose/issues/new/choose

**Comment by Uzlopak** [comment]
You are invited to generate new issues to test the changes in my fork ;)

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  
+  - type: input
+    id: mongo-version
+    attributes:
+      label: MongoDB version
+      placeholder: 5.x
+    validations:
+      required: true
+
+  - type: dropdown
```
Comment (vkarpov15) [inline .github/ISSUE_TEMPLATE/bug.yml:40]: Let's get rid of the OS input, this is rarely significant

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    attributes:
+      label: Operating system
+      options:
+        - Linux
+        - macOS
+        - Windows
+    validations:
+      required: false
+
+  - type: input
```
Comment (vkarpov15) [inline .github/ISSUE_TEMPLATE/bug.yml:51]: Same, let's get rid of this one

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+blank_issues_enabled: false
+documentation:
+  - name: 📘 Documentation
+    url: https://mongoosejs.com/docs/guide.html
+    about: The manual of mongoose
```
Comment (vkarpov15) [inline .github/ISSUE_TEMPLATE/config.yml:5]: "The manual of mongoose" sounds a bit strange. Let's just put "Mongoose docs"

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    attributes:
+      label: Node.js version
+      placeholder: 14.x
+    validations:
+      required: true
+  
+  - type: input
+    id: mongo-version
+    attributes:
+      label: MongoDB version
```
Comment (vkarpov15) [inline .github/ISSUE_TEMPLATE/bug.yml:35]: To make this more clear, let's put "MongoDB server version".

Many times, people put their version of the [mongodb npm package](https://npmjs.com/package/mongodb), which isn't helpful.

File: .github/ISSUE_TEMPLATE/regression.yml [hunk 5]
```diff
@@ -0,0 +1,80 @@
+name: 💥 Regression Report
```
Comment (vkarpov15) [inline .github/ISSUE_TEMPLATE/regression.yml:1]: Why separate "bug" and "regression"?

**Review by vkarpov15 — COMMENTED** [review]
A couple of comments, but otherwise this is great :+1:

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 1 above)
Comment (Uzlopak) [reply .github/ISSUE_TEMPLATE/bug.yml]: done

File: .github/ISSUE_TEMPLATE/config.yml (hunk 3 above)
Comment (Uzlopak) [reply .github/ISSUE_TEMPLATE/config.yml]: done

File: .github/ISSUE_TEMPLATE/regression.yml (hunk 5 above)
Comment (Uzlopak) [reply .github/ISSUE_TEMPLATE/regression.yml]: removed

**Comment by Uzlopak** [comment]
@vkarpov15 
Can I merge this?

**Review by vkarpov15 — APPROVED** [review]

**Comment by vkarpov15** [comment]
@Uzlopak thanks for your patience :+1:

**Pull request merged**